cirq/google/api/v2/program.proto

```diff
@@ -133,10 +137,15 @@ message Arg {
   //
   // Functions are used to define a simple s-expression tree describing
   // how to combine numbers and symbols mathematically.
+  //
+  // Reference indices are used to avoid repeating long strings in the
+  // serialized proto.  This is a number that refers to the index of
+  // the reference string in the program.
   oneof arg {
     ArgValue arg_value = 1;
     string symbol = 2;
     ArgFunction func = 3;
+    int32 reference_index = 4;
```

Comment:
Might want to call the underlying table `symbols` and the field here `symbol_index`; that makes clear that this is an alternative to specifying `symbol` as a string directly. It's actually kind of nice to have a single top-level "symbol table" where we can look to see all symbols that need to be resolved for a given program.